src/plugins/kibana/public/dashboard/components/panel/panel.js

```diff
@@ -55,6 +55,9 @@ uiModules
           // create child ui state from the savedObj
           const uiState = panelConfig.uiState || {};
           $scope.uiState = $scope.parentUiState.createChild(getPanelId(panelConfig.panel), uiState, true);
+          if (panelConfig.setUiState) {
```

Comment:
See my note in `panel/lib/visualization.js`, I think this logic should be moved there.